src/_common/video/player/controller.ts

```diff
@@ -53,31 +62,56 @@ export function toggleVideoPlayback(player: VideoPlayerController) {
 	}
 }
 
-export function setVideoVolume(player: VideoPlayerController, level: number) {
+export function setVideoVolume(
+	player: VideoPlayerController,
+	level: number,
+	storeLevel: boolean,
+	forceUnmute = true
+) {
 	const volume = Math.min(1, Math.max(0, Math.round(level * 100) / 100));
+	if (volume && storeLevel) {
+		player.mutedFallbackVolume = volume;
+	}
 
-	// Assign volume level to the proper local storage context.
-	switch (player.context) {
-		case 'feed':
-			SettingVideoPlayerFeedVolume.set(volume);
-			break;
-		case 'page':
-			SettingVideoPlayerVolume.set(volume);
-			break;
+	if (player.context === 'feed') {
+		SettingVideoPlayerFeedVolume.set(volume);
+	} else if (player.context === 'page') {
+		SettingVideoPlayerVolume.set(volume);
 	}
 
 	player.volume = volume;
+
+	if (volume && player.isMuted && forceUnmute) {
+		toggleVideoMuted(player);
+	}
+
+	if (!volume && !player.isMuted) {
+		toggleVideoMuted(player);
+	}
+}
+
+export function toggleVideoMuted(player: VideoPlayerController) {
+	player.isMuted = !player.isMuted;
+	if (!player.isMuted && player.mutedFallbackVolume !== null) {
+		setVideoVolume(player, player.mutedFallbackVolume, false);
+	}
+
+	if (player.context === 'feed') {
+		SettingVideoPlayerFeedMuted.set(player.isMuted);
+		// Feed videos don't have a volume slider, so if the video element volume was somehow changed we want to set it back to 100%.
+		if (!player.isMuted) {
```

Comment:
I think you should move this condition back up into the first if of the function:
```
if (!player.isMuted) {
    if (player.mutedFallbackVolume) {
        setVideoVolume(player, player.mutedFallbackVolume);
    } else if (player.context === 'feed') {
        setVideoVolume(player, 100);
    }
}
```